alacritty/src/logging.rs

```diff
@@ -63,55 +94,32 @@ impl log::Log for Logger {
     }
 
     fn log(&self, record: &log::Record<'_>) {
-        if self.enabled(record.metadata()) && record.target().starts_with("alacritty") {
-            let now = time::strftime("%F %T.%f", &time::now()).unwrap();
-
-            let msg = if record.level() >= Level::Trace {
-                format!(
-                    "[{}] [{}] [{}:{}] {}\n",
-                    now,
-                    record.level(),
-                    record.file().unwrap_or("?"),
-                    record.line().map(|l| l.to_string()).unwrap_or_else(|| "?".into()),
-                    record.args()
-                )
-            } else {
-                format!("[{}] [{}] {}\n", now, record.level(), record.args())
-            };
-
-            if let Ok(ref mut logfile) = self.logfile.lock() {
-                let _ = logfile.write_all(msg.as_ref());
-
-                if record.level() <= Level::Warn {
-                    #[cfg(not(windows))]
-                    let env_var = format!("${}", ALACRITTY_LOG_ENV);
-                    #[cfg(windows)]
-                    let env_var = format!("%{}%", ALACRITTY_LOG_ENV);
-
-                    let msg = format!(
-                        "[{}] See log at {} ({}):\n{}",
-                        record.level(),
-                        logfile.path.to_string_lossy(),
-                        env_var,
-                        record.args(),
-                    );
-                    let message_type = match record.level() {
-                        Level::Error => MessageType::Error,
-                        Level::Warn => MessageType::Warning,
-                        _ => unreachable!(),
-                    };
-
-                    if let Ok(event_proxy) = self.event_proxy.lock() {
-                        let mut message = Message::new(msg, message_type);
-                        message.set_target(record.target().to_owned());
-
-                        let _ = event_proxy.send_event(Event::Message(message));
-                    }
-                }
-            }
+        // Get target crate.
+        let index = record.target().find(':').unwrap_or_else(|| record.target().len());
+        let target = &record.target()[..index];
+
+        // Only log our own crates.
+        if !self.enabled(record.metadata())
+            || !matches!(target, "alacritty_terminal" | "alacritty" | "crossfont")
+        {
+            return;
+        }
```

Comment:
This check in inaccurate. We have `alacritty_config` as a target for config errors, etc, so you should likely add it here as well, since `target` is not a crate name.

Right now it leads to config errors not being displayed in the message bar, you can test with `alt_send_esc: true1`.

Also, I'm not sure whether crossfont errors should reach message bar tbqh. I'm fine with seeing them in a log output of `-vvv`, but I don't want them to populate in message bar, I guess, since I feel like it could spam message a lot, and since we don't have close all functionality it's hard to delete things in it.

So, I'd say `alacritty_config` and `alacritty` should be the only target we could show in a message bar. What do you think?

Reply:
I've considered if we should have something like crates which log to the logfile vs crates allowed to log to the message bar. For something like winit it might be useful but since we have control over crossfont I think we should try to make messages warning/errors only if they are critical enough that a user might want to see them.

I've checked out the code of crossfont when I wrote this and the only warning+ it uses is when a font cannot be found on macOS, which would be something that could be helpful for the user for example.

Reply:
Then it's likely fine, but I feel like we log error in alacritty as well if we fail to load something(you've changed that code recently)?

Reply:
I've mentioned this on IRC but just to document this: Hitting the only `warn!` message in `crossfont` does not cause any error propagation. There are no `error!` logs. So there should not be any duplicate messages.